### ISSUE TITLE
dedicated -dec or -fac option for tensor2metric (feature request)

### ISSUE THREAD

**Issue by mschira** [MRtrix3/mrtrix3#2175]
Currently there exists a -vec option to extract the vector directions, but this results in negative and positive values.
So the files generated will not be displayed correctly in many external tools, for example ITKSnap.

A -dec or -fac option could do the same as -vec, but only take absolute values. This would save a subsequence operation, and most importantly make this pitfall more obvious especially for beginners or casual users.
M.

**Comment by bjeurissen** [comment]
Thanks for your suggestion! Like you said: this operation can be performed simply by piping to ``mrcalc IN -abs OUT``, so the added value is somewhat limited. Moreover, the resulting files will still not be compatible with most third-party viewers: the directions saved by ``dwi2tensor -vec`` are represented in scanner coordinates (as all directions information MRtrix), so the resulting values of abs will not be straight RGB values that can be used by other software packages...

**Comment by mschira** [comment]
Yes, not debating the limited scope - so would presumably be the effort.

I think the benefit would be more in the easier to understand department. Avoiding confusion in particular for casual users.
I know what I am talking about. This has just cost me a significant part of my Monday until I figured out that my DWI, bvec and other things were perfectly fine.
It was only when I loaded the, let's call them vector maps, into matlab that I discovered that there are negative RGB values (or not RGB...) and making the abs and saving was all required to turn them into perfect FAC maps.

Alternatively, a modest improvement of the help text for the -vector

"-vector image compute the selected eigenvector(s) of the diffusion tensor. Will have negative vectors, for FAC values, take the abs of this output."

something along this line.
Cheers
Mark

**Comment by Lestropie** [comment]
Alternatively / in addition, commands now give the opportunity to define example usages; so demonstrating `tensor2metric -vector - | mrcalc - -abs` can be achieved explicitly within the `tensor2metric` help page. I generally prefer to do this rather than giving long explanations within the help text of individual command-line options, as the latter extends and separates out the command-line option list more than I like. Though the `-vector` option could still have a generic statement, e.g. that the representation is as a direction on the sphere in scanner coordinates or something like that.
